### PR TITLE
Combine Drone release steps

### PR DESCRIPTION
**Pull request by silverwind**
See https://github.com/go-gitea/gitea/issues/9324. I don't know if this will fix the issue but it seems worth a try to land this on master just to watch the release-master task run. If it does not work out, we can revert it.

### CODE REVIEW

**Comment by lafriks** [comment]
I don't see how this does change things as checked out directory are shared between steps except if there are not in the same pipeline. Also imho js/css build step should be in different drone step

**Comment by silverwind** [comment]
Yes, they should be shared but the only logical explanation for https://github.com/go-gitea/gitea/issues/9324 is that they aren't for some reason.

I do think it should be possible to produce a working release locally via `make release` and not only via CI, so I see the `release` task dependencies as must-haves.

**Comment by techknowlogick** [comment]
So embedding works (we can see this with vendor'd css/js, as well as templates are embedded, this means go generate with bindata env is working.

I'm hesitant to merge this without testing, luckily drone allows us to run specific pipelines via CLI. I will test this change via CLI (which will allow me to also put in some debugging into drone that'll show more info (ex. `ls` of a specific dir before running `make release`)

**Comment by zeripath** [comment]
This change means we lose testing on node10 doesn't it?

File: .drone.yml [hunk 1]
```diff
@@ -389,16 +389,11 @@ steps:
       - git fetch --tags --force
 
   - name: build
```
Comment (techknowlogick) [inline .drone.yml]:
```suggestion
   - name: static
```
There are steps below that depend on this being the step name

File: .drone.yml [hunk 2]
```diff
@@ -497,16 +492,11 @@ steps:
...
```
Comment (techknowlogick) [inline .drone.yml]:
```suggestion
   - name: static
```

**Review by techknowlogick — COMMENTED** [review]
Just ran `drone exec --pipeline release-master` with the changes I suggested to this PR and it builds as expected.

Thank you for this PR @silverwind :)

**Comment by techknowlogick** [comment]
@zeripath nope, this is just for the release steps, the node10 container is still in the testing pipeline.

**Comment by axifive** [comment]
@techknowlogick Can you also debugging already existing step in master? To understand why it's not worked, maybe this can help avoid similar errors in the future.

**Comment by techknowlogick** [comment]
Oh, I know what's going on! I can't believe I missed it. The build & static step are run at the same time, this means that even though the files exist where they should, the go generate command ends up being run before the JS/CSS files are finished being generated. the "depends_on" needs to be removed from each step in the release pipelines to force each step to be run in sequence.

**Review by techknowlogick — CHANGES_REQUESTED** [review]
Blocking per my most recent comment.

**Comment by silverwind** [comment]
@techknowlogick thanks for finding it. It's now clear to me that as soon as a `depends_on` is introduced, drone will parallelize (https://docker-runner.docs.drone.io/configuration/parallelism/). I went ahead and removed all `depends_on` in the release pipelines so they should now always serialize.

**Review by zeripath — APPROVED** [review]

**Comment by silverwind** [comment]
Squashed and added new commit message.

**Comment by silverwind** [comment]
One more tweak pushed which make `js css` dependencies of `generate`. This fixes the case where`make generate build` (which some people have in their build scripts) would result in `generate` running only once and too early.

**Comment by silverwind** [comment]
One more tweak: Removed useless `go-all` target. Now `go-check` only runs once instead of twice and before `generate` which also depends on `go` present.

The `go` target itself could potentially also removed but I kept it because it allows for quicker building without `generate`. Thought, the same could also be achieved using `make gitea`.

**Comment by silverwind** [comment]
`go` target removed as well for simplicity's sake. One can still run `make gitea` (`make gitea.exe` on Windows) to achieve the same.

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
Thanks @silverwind for all of your work on this :)

**Pull request merged**